client/src/main.tsx

```diff
@@ -25,15 +28,27 @@ const router = createBrowserRouter([
         });
 
         const parsedResponse = userProfileSchema.safeParse(response.json());
-        if (!parsedResponse.success) {
-          console.log("Unable to make implicit login");
-          return redirect("/login");
-        }
-        return <Outlet />;
+        if (!parsedResponse.success) throw new Error("Unable to make implicit login");
+
+        if ("errorMessage" in parsedResponse.data) throw new Error("Error");
+
+        useProfileStore.setState(parsedResponse.data);
+        resolve("Implicit login completed");
       } catch (error) {
-        console.log("Unexpected error");
-        return redirect("/login");
+        reject(error);
       }
+    }, 500);
+  });
+};
+
+const router = createBrowserRouter([
+  {
+    path: "/",
+    loader: async () => {
```

Comment:
This function doesn't need to be `async`, and we can simplify it

I recommend something like:
```typescript
loader: () => defer({ response: implicitLogin() }),
```